packages/web/src/components/track/GiantTrackTile.tsx

```diff
@@ -0,0 +1,655 @@
+import { useCallback, useState } from 'react'
+
+import {
+  squashNewLines,
+  getCanonicalName,
+  formatDate,
+  formatSeconds,
+  Genre,
+  FeatureFlags,
+  isAudiusUrl,
+  getPathFromAudiusUrl,
+  Nullable,
+  Remix,
+  CoverArtSizes,
+  ID,
+  PremiumConditions,
+  FieldVisibility,
+  getDogEarType
+} from '@audius/common'
+import {
+  Button,
+  ButtonType,
+  IconShare,
+  IconRocket,
+  IconRepost,
+  IconHeart,
+  IconKebabHorizontal
+} from '@audius/stems'
+import cn from 'classnames'
+import Linkify from 'linkify-react'
+
+import { ReactComponent as IconRobot } from 'assets/img/robot.svg'
+import { ArtistPopover } from 'components/artist/ArtistPopover'
+import DownloadButtons from 'components/download-buttons/DownloadButtons'
+import { EntityActionButton } from 'components/entity-page/EntityActionButton'
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+import Menu from 'components/menu/Menu'
+import RepostFavoritesStats from 'components/repost-favorites-stats/RepostFavoritesStats'
+import { SearchTag } from 'components/search/SearchTag'
+import Skeleton from 'components/skeleton/Skeleton'
+import { Tile } from 'components/tile'
+import Toast from 'components/toast/Toast'
+import Tooltip from 'components/tooltip/Tooltip'
+import { ComponentPlacement } from 'components/types'
+import UserBadges from 'components/user-badges/UserBadges'
+import { getFeatureEnabled } from 'services/remote-config/featureFlagHelpers'
+import { moodMap } from 'utils/Moods'
+
+import { AiTrackSection } from './AiTrackSection'
+import Badge from './Badge'
+import { CardTitle } from './CardTitle'
+import GiantArtwork from './GiantArtwork'
+import styles from './GiantTrackTile.module.css'
+import { GiantTrackTileProgressInfo } from './GiantTrackTileProgressInfo'
+import InfoLabel from './InfoLabel'
+import { PlayPauseButton } from './PlayPauseButton'
+import { PremiumTrackSection } from './PremiumTrackSection'
+
+const BUTTON_COLLAPSE_WIDTHS = {
+  first: 1095,
+  second: 1190,
+  third: 1286
+}
+
+// Toast timeouts in ms
+const REPOST_TIMEOUT = 1000
+const SAVED_TIMEOUT = 1000
+
+const messages = {
+  makePublic: 'MAKE PUBLIC',
+  isPublishing: 'PUBLISHING',
+  repostButtonText: 'repost',
+  repostedButtonText: 'reposted',
+  unplayed: 'Unplayed',
+  timeLeft: 'left',
+  played: 'Played',
+  generatedWithAi: 'Generated With AI',
+  actionGroupLabel: 'collection actions'
+}
+
+export type GiantTrackTileProps = {
```

Comment:
These were all converted from `GiantTrackTile.propTypes`. However, that type just used things like `object` and `string` for some props which have more explicit types (i.e. `Remix`, `PremiumConditions`, `CoverArtSizes`).  Pretty much anything in here that isn't a number/string/boolean is now more strongly typed.